### PR TITLE
Adding ca-certificate parameter in tcbuild.yaml.

### PR DESCRIPTION
**Pull request by franc101**
Adding ca-certificate parameter in tcbuild.yaml for use private docker registry.

### CODE REVIEW

File: tcbuilder/backend/tcbuild.schema.yaml [hunk 1]
```diff
@@ -305,6 +305,9 @@ properties:
                   registry:
                     type: string
                     description: "Alternative container registry used to access container image"
+                  ca-certificate:
+                    type: string
+                    description: "CA certificate path of alternative container registry"
```
Comment (microhobby) [inline tcbuilder/backend/tcbuild.schema.yaml:310]: I think should good add also a `tdxMeta` property, these are used to help integrated development environments know what show in the autocompletion. For this case I think that the IDE should show the `*.cer` files, right? So, maybe something like this:

```yaml
tdxMeta: "file;cer"
```

**Comment by Nathan-ma** [comment]
Hi @franc101,

We appreciate the contribution. The changes look ok to me, but for any changes in torizoncore-builder to be accepted, we require a test case for them to be written.

I kindly ask you to write a test case, testing these changes in the `tests/integration/testcases/build.bats`; add them to this PR, manually run the test cases for the build command and share the report's content in a comment.

Another critical test is the python linter. Please also add the linter test results in a comment.

To run the test cases for the build command:
```
# in the repo's main folder
$ bash << 'EOF'
docker build -f torizoncore-builder.Dockerfile -t torizoncore-builder:local .
cd tests/integration &&  mkdir -p workdir/images
wget https://artifacts.toradex.com:443/artifactory/torizoncore-oe-prerelease-frankfurt/dunfell-5.x.y/monthly/24/apalis-imx6/torizon-upstream/torizon-core-docker/oedeploy/torizon-core-docker-apalis-imx6-Tezi_5.7.0-devel-202206%2Bbuild.24.tar -P workdir/images && touch workdir/images/.images_downloaded
source setup.sh
TCB_CUSTOM_IMAGE=torizoncore-builder:local TCB_REPORT=1  TCB_TESTCASE=build ./run.sh
EOF
```

After running this command, a test report located at `tests/integration/workdir/reports/` will be generated. This is the content you'll need to share in the comment.

---

To run the python linter:
```
# in the repo's main  folder
$ docker build -f torizoncore-builder.Dockerfile --target tcbuilder-dev -t torizoncore-builder-dev:local .
$ alias torizoncore-tools='docker run --rm -it -v /deploy -v $(pwd):/workdir -v storage:/storage \
--net=host -v /var/run/docker.sock:/var/run/docker.sock torizoncore-builder-dev:local'
$ torizoncore-tools bash -c 'cd /workdir && pylint -ry --output-format=colorized \
--disable=missing-module-docstring \ 
--disable=missing-class-docstring \ 
--disable=missing-function-docstring \ 
--disable=fixme \ 
 $(find tcbuilder/ -type f -name "*.py") *.py'
```

Thank you!

File: tcbuilder/backend/tcbuild.schema.yaml (hunk 1 above)
Comment (franc101) [reply tcbuilder/backend/tcbuild.schema.yaml:310]: I have not found any information about txdMeta format. For ```*.cer``` and ```*.crt``` files, is ```tdxMeta: "file;cer,crt"``` ok?

Comment (microhobby) [reply tcbuilder/backend/tcbuild.schema.yaml:310]: Hi @franc101 yeah this is not documented yet, but you are right `file;cer,crt` should work to show the completions for all `*.cer` and `*.crt` in the workspace. Thanks.

**Comment by franc101** [comment]
Hi @Nathan-ma,

This is the report's content after run the test cases with the new one based on bundle's test cases.

```
1..12
ok 1 build: check help output in 1021ms
ok 2 build: template creation in 2230ms
ok 3 build: config file with error detected by YAML parser in 913ms
ok 4 build: config file with multiple validation errors in 971ms
ok 5 build: config file with variables in 2116ms
ok 6 build: config file with variables in input section in 1267ms
ok 7 build: full customization checked on host in 27711ms
ok 8 build: full customization checked on device # skip device not configured
ok 9 build: config file with autoinstall and autoreboot in 41551ms
ok 10 build: check overlays's clear in 98605ms
ok 11 build: basic tcbuild referencing a docker-compose file in 37413ms
ok 12 build: check with secure registry with authentication in 45682ms
```

**Comment by Nathan-ma** [comment]
Hi @franc101,

Thank you for the contribution!  You can consider this PR approved.

The test case and changes looks ok to me, I will squash and add the commits on this branch to our main branch soon.

One last thing I would like to ask from you if you don't mind, could you sign one of your commits? A simple commit command with the `-s` option will be enough. You can even amend the signature to the last commit and force push if you want to.

Once again, I appreciate the contribution.

**Pull request merged**